docs/styleguide/PyGuide.md

```diff
@@ -0,0 +1,820 @@
+
+    def transform(self):
+        """
+        Applies transformations to the Keras model.
+
+        :return: transformed Keras model
+        """
+```
+
+<a id="s3.5.4-block-and-inline-comments"></a>
+<a id="354-block-and-inline-comments"></a>
+<a id="block-and-inline-comments"></a>
+#### 3.5.4 Block and Inline Comments 
+
+The final place to have comments is in tricky parts of the code. If you're going to have to explain it 
+in the future, you should comment it now. Complicated operations get a few lines of comments before 
+the operations commence. Non-obvious ones get comments at the end of the line.
+
+```python
+# We use a weighted dictionary search to find out where i is in
+# the array.  We extrapolate position based on the largest num
+# in the array and the array size and then do binary search to
+# get the exact number.
+
+if i & (i-1) == 0:  # True if i is 0 or a power of 2.
+```
+
+To improve legibility, these comments should start at least 2 spaces away from
+the code with the comment character `#`, followed by at least one space before
+the text of the comment itself.
+
+On the other hand, never describe the code. Assume the person reading the code
+knows Python (though not what you're trying to do) better than you do.
+
+```python
+# BAD COMMENT: Now go through the b array and make sure whenever i occurs
+# the next element is i+1
+```
+
+<a id="s3.6-strings"></a>
+<a id="36-strings"></a>
+<a id="strings"></a>
+### 3.6 Strings 
+
+Use 'something' instead of "something"
+
+```python
+# Correct:
+
+long_string = '''This is fine if your use case can accept
+    extraneous leading spaces.'''
+
+long_string = ('And this is fine if you cannot accept\n' +
+               'extraneous leading spaces.')
+
+long_string = ('And this too is fine if you cannot accept\n'
+               'extraneous leading spaces.')
+
+import textwrap
+
+long_string = textwrap.dedent('''\
+    This is also fine, because textwrap.dedent()
+    will collapse common leading spaces in each line.''')
+```
+
+<a id="s3.7-logging"></a>
+<a id="37-logging"></a>
+<a id="logging"></a>
+### 3.7 Logging 
+
+For logging functions that expect a pattern-string (with %-placeholders) as
+their first argument: Always call them with a string literal (not an f-string!)
+as their first argument with pattern-parameters as subsequent arguments. Some
+logging implementations collect the unexpanded pattern-string as a queryable
+field. It also prevents spending time rendering a message that no logger is
+configured to output.
+
+```python
+# Correct:
+import nncf
+from nncf.common.utils.logger import logger as nncf_logger
+nncf_logger.info('Test message: %s', nncf.__version__)
+```
+
+<a id="s3.8-error-messages"></a>
+<a id="38-error-messages"></a>
+<a id="error-messages"></a>
+### 3.8 Error Messages 
+
+Error messages (such as: message strings on exceptions like `ValueError`, or
+messages shown to the user) should follow three guidelines:
```

Comment:
Three? Only two were provided.

I think it is necessary to add that the message should start with a capital letter.